### PR TITLE
kie-issues#1320: Fix and add environment variables to the dev-deployment-dmn-form-webapp

### LINKED ISSUE #1320: Create an environment variable to map the backend service for `dev-deployment-dmn-form-webapp-image`
- Create an environment variable that hosts the URL of the quarkus app serving the backend for the form;
- Add this env var to the image (Containerfile);
- Add a description and instructions for it in the README.

### PR DESCRIPTION
**Pull request by thiagoelg**
Closes https://github.com/apache/incubator-kie-issues/issues/1320

In this PR:
- Fixed an issue where the dev-deployment-dmn-form-webapp wouldn't process the quarkus OpenAPI spec correctly;
- Fixed the dev-webapp for the dev-deployment-dmn-form-webapp;
- Added new environment variables to configure dev-deployment-dmn-form-webapp;
- Upgraded registry.access.redhat.com/ubi9/openjdk-17 image to tag 1.20;

### CODE REVIEW

File: packages/dev-deployment-dmn-form-webapp/static/env.json
Comment (ljmotta) [inline packages/dev-deployment-dmn-form-webapp/static/env.json:1]: Shouldn't it have the apache header?

File: packages/dev-deployment-dmn-form-webapp/src/DmnDevDeploymentFormWebAppDataApi.tsx [hunk 1]
```diff
@@ -30,38 +28,51 @@ export interface FormData {
...
-    JSON.stringify(openApiSpec).replace(
-      new RegExp(`${args.basePath ? `/${args.basePath}` : ""}/dmnDefinitions.json`, "g"),
-      routes.quarkusApp.dmnDefinitionsJson.path({}, args.baseOrigin, args.basePath)
-    )
-  );
 
-  const dereferencedSpec = await OpenAPIParser.dereference(fixedRefOpenApiSpec, {
+  // Append origin to schema $refs
+  Object.keys(openApiSpec.paths).forEach((modelPath) => {
+    console.log(modelPath);
```
Comment (ljmotta) [inline packages/dev-deployment-dmn-form-webapp/src/DmnDevDeploymentFormWebAppDataApi.tsx]: leftover?

File: packages/dev-deployment-dmn-form-webapp-image/README.md [hunk 2]
```diff
@@ -15,7 +15,39 @@
    under the License.
 -->
 
-## @kie-tools/dev-deployment-dmn-form-webapp-image
+# KIE Sandbox Dev Deployment DMN Form Webapp Image
+
+This image is ready to be used for Dev deployments that contain Decisions (`.dmn`) on KIE Sandbox.
```
Comment (ljmotta) [inline packages/dev-deployment-dmn-form-webapp-image/README.md]:
```suggestion
This image is ready to be used for Dev Deployments that contain Decisions (`.dmn`) on KIE Sandbox.
```

File: packages/dev-deployment-dmn-form-webapp-image/README.md [hunk 3]
```diff
@@ -15,7 +15,39 @@
...
+It expects a Quarkus application to be running in the same host, which it will use to fetch information to render a form that let's users interact with a Decision.
+
+For example, if this image is running on `localhost:8080/form-webapp` it will try to fetch the OpenAPI specification from `localhost:8080/q/openapi` and use that to map all other routes and endpoints for the Quarkus application, plus all of the form inputs necessary for each DMN.
+
+Unfortunatelly, with this condition, is pratically impossible to run these container and a Quarkus app locally, because it's impossible to run two containers on the same port.
+To solve this issue, two environment variables were created to configure this behavior:
+
+- DEV_DEPLOYMENT_DMN_FORM_WEBAPP_QUARKUS_APP_ORIGIN: Sets the origin of the quarkus app (`http://localhost:8080` for example); Defaults to `""`, in which case it's replaced by `window.location.origin`.
```
Comment (ljmotta) [inline packages/dev-deployment-dmn-form-webapp-image/README.md]:
```suggestion
- DEV_DEPLOYMENT_DMN_FORM_WEBAPP_QUARKUS_APP_ORIGIN: Sets the origin of the Quarkus app (`http://localhost:8080` for example); Defaults to `""`, in which case it's replaced by `window.location.origin`.
```

File: packages/dev-deployment-dmn-form-webapp-image/README.md [hunk 4]
```diff
@@ -15,7 +15,39 @@
...
+- DEV_DEPLOYMENT_DMN_FORM_WEBAPP_QUARKUS_APP_PATH: Sets the relative path of the quarkus app in that origin (`/dev-deployment-1234` for example); Defaults to `""`, in which case it's replaced by `..` or `/` depending on the path the webapp is being served from.
```
Comment (ljmotta) [inline packages/dev-deployment-dmn-form-webapp-image/README.md]:
```suggestion
- DEV_DEPLOYMENT_DMN_FORM_WEBAPP_QUARKUS_APP_PATH: Sets the relative path of the Quarkus app in that origin (`/dev-deployment-1234` for example); Defaults to `""`, in which case it's replaced by `..` or `/` depending on the path the webapp is being served from.
```

File: packages/dev-deployment-dmn-form-webapp-image/README.md [hunk 5]
```diff
@@ -15,7 +15,39 @@
...
+
+To develop this webapp you can start the webapp container and a quarkus app container to be used as backend:
+
+```bash
+docker run -d -p 8080:8080 -e DEV_DEPLOYMENT__UPLOAD_SERVICE_API_KEY=dev docker.io/apache/incubator-kie-sandbox-dev-deployment-kogito-quarkus-blank-app:main
+
+docker run -d -p 8081:8081 -e DEV_DEPLOYMENT_DMN_FORM_WEBAPP_QUARKUS_APP_ORIGIN=http://localhost:8080 docker.io/apache/incubator-kie-sandbox-dev-deployment-dmn-form-webapp:main
+```
+
+The quarkus app will be running on port 8080 and you will have to upload some DMNs to it (check the [dev-deployment-kogito-quarkus-blank-app-image README](../dev-deployment-kogito-quarkus-blank-app-image/README.md));
```
Comment (ljmotta) [inline packages/dev-deployment-dmn-form-webapp-image/README.md]:
```suggestion
The Quarkus app will be running on port 8080 and you will have to upload some DMNs to it (check the [dev-deployment-kogito-quarkus-blank-app-image README](../dev-deployment-kogito-quarkus-blank-app-image/README.md));
```

File: packages/dev-deployment-dmn-form-webapp-image/README.md [hunk 6]
```diff
@@ -15,7 +15,39 @@
...
+
+The DMN Form Webapp will be running on port 8081 and will be using the quarkus app on port 8080 as it's backend.
```
Comment (ljmotta) [inline packages/dev-deployment-dmn-form-webapp-image/README.md]:
```suggestion
The DMN Form Webapp will be running on port 8081 and will be using the Quarkus app on port 8080 as it's backend.
```

**Review by ljmotta — COMMENTED** [review]
@thiagoelg Thanks for the PR. I've just made a few minor comments inline.

File: packages/dev-deployment-dmn-form-webapp-image/README.md (hunk 2 above)
Comment (thiagoelg) [reply packages/dev-deployment-dmn-form-webapp-image/README.md]: This is something mixed up in the code base (my fault). We have `Dev deployments` and `Dev Deployments` all over the place.

I'm not sure how to capitalize it, but I'm siding with the `Dev Deployments` option, this way we can set a pattern of: "Feature names should be capitalized while concepts should not."

WDYT?

File: packages/dev-deployment-dmn-form-webapp/src/DmnDevDeploymentFormWebAppDataApi.tsx (hunk 1 above)
Comment (thiagoelg) [reply packages/dev-deployment-dmn-form-webapp/src/DmnDevDeploymentFormWebAppDataApi.tsx]: 🙈

File: packages/dev-deployment-dmn-form-webapp/static/env.json
Comment (thiagoelg) [reply packages/dev-deployment-dmn-form-webapp/static/env.json:1]: Actually, it shouldn't. This is a JSON file, we can't add comments on JSON files.

**Review by ljmotta — APPROVED** [review]

**Review by tiagobento — APPROVED** [review]

**Pull request merged**